apps/native-component-list/src/screens/Video/VideoFlashListScreen.tsx

```diff
@@ -0,0 +1,110 @@
+import AntDesign from '@expo/vector-icons/AntDesign';
+import { FlashList } from '@shopify/flash-list';
+import { VideoView, VideoSource, useVideoPlayer } from 'expo-video';
+import { useEffect, useState } from 'react';
+import { View } from 'react-native';
+
+import { styles, ViewSize } from './VideoFlatListScreen';
+import { localVideoSource } from './videoSources';
+
+const videoSources = Array(100).fill(localVideoSource);
+
+type RenderItemProps = {
+  item: VideoSource;
+  index: number;
+  viewSize: ViewSize;
+  visibleIndex: number;
+};
+
+function RenderItem({ item, index, viewSize, visibleIndex }: RenderItemProps) {
+  const [heart, setHeart] = useState(false);
+  // We don't pass the item directly into the player, when the item changes in the argument
+  // the video player is re-created, which is slow
+  const player = useVideoPlayer(null, (player) => {
+    player.bufferOptions = {
+      preferredForwardBufferDuration: 5,
+    };
+    player.loop = true;
+    player.pause();
+  });
+
+  // Instead, we use the replace function
+  useEffect(() => {
+    player.replace(item);
+  }, [index]);
```

Comment:
Player dependency is missing. Can be silenced?

Reply:
@hirbod Yep, the player won't change anyways.